packages/aws-cdk/lib/api/accelerated-deployments.ts

```diff
@@ -0,0 +1,242 @@
+import { ISDK, Mode, SdkProvider } from './aws-auth';
+import { DeployStackResult } from './deploy-stack';
+import { CloudFormationStack } from './util/cloudformation';
+import { evaluateCfn } from './util/cloudformation/evaluate-cfn';
+
+/**
+ * Perform an accelerated deployment,
+ * short-circuiting CloudFormation if possible.
+ * If it's not possible to short-circuit the deployment
+ * (because the CDK Stack contains changes that cannot be deployed without CloudFormation),
+ * returns `undefined`.
+ */
+export async function performAcceleratedDeployment(
+  sdkProvider: SdkProvider, assetParams: { [key: string]: string },
+  cloudFormationStack: CloudFormationStack, stackArtifact: cxapi.CloudFormationStackArtifact,
+): Promise<DeployStackResult | undefined> {
+  const currentTemplate = await cloudFormationStack.template();
+  const stackChanges = cfn_diff.diffTemplate(currentTemplate, stackArtifact.template);
+
+  // resolve the environment, so we can substitute things like AWS::Region in the CFN expressions
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  const acceleratableChanges = findAllAcceleratableChanges(stackChanges, {
+    ...assetParams,
+    'AWS::Region': resolvedEnv.region,
+    'AWS::AccountId': resolvedEnv.account,
+  });
+  if (!acceleratableChanges) {
+    // this means there were changes to the template that cannot be short-circuited
+    return undefined;
+  }
+
+  // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
+  // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
+  const sdk = await sdkProvider.forEnvironment(resolvedEnv, Mode.ForWriting);
+  // apply the short-circuitable changes
+  await applyAllAcceleratedChanges(sdk, stackArtifact, acceleratableChanges);
+
+  return { noOp: acceleratableChanges.length === 0, stackArn: cloudFormationStack.stackId, outputs: cloudFormationStack.outputs, stackArtifact };
+}
+
+function findAllAcceleratableChanges(
+  stackChanges: cfn_diff.TemplateDiff, assetParamsWithEnv: { [key: string]: string },
+): LambdaFunctionResource[] | undefined {
+  const acceleratableResources = new Array<LambdaFunctionResource>();
+  let foundNonAcceleratableChange = false;
+  stackChanges.resources.forEachDifference((logicalId: string, change: cfn_diff.ResourceDifference) => {
+    const lambdaFunctionShortCircuitChange = isAcceleratedLambdaFunctionChange(logicalId, change, assetParamsWithEnv);
+    if (lambdaFunctionShortCircuitChange === false) {
+      foundNonAcceleratableChange = true;
+    } else if (lambdaFunctionShortCircuitChange === true) {
+      // empty 'if' just for flow-aware typing to kick in...
+    } else {
+      acceleratableResources.push(lambdaFunctionShortCircuitChange);
+    }
+  });
+  return foundNonAcceleratableChange ? undefined : acceleratableResources;
+}
+
+function isAcceleratedLambdaFunctionChange(
```

Comment:
can you move the lambda-related functions to a separate file under a directory called `hot-swap/` - then, each resource will have their own file there.

Reply:
Done.